docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -8,48 +8,141 @@ title: Authorization
 
 ### 1.1 Purpose and Scope
 
-The Model Context Protocol provides authorization capabilities at the transport level,
-enabling MCP clients to make requests to restricted MCP servers on behalf of resource
-owners. This specification defines the authorization flow for HTTP-based transports.
-
-### 1.2 Protocol Requirements
+The Model Context Protocol provides authorization capabilities at the transport level, enabling MCP
+clients to make requests to protected MCP servers on behalf of resource owners. This specification
+defines the authorization flow for HTTP-based transports.
+
+The scope of this document is **user authorization** - how a MCP client can authenticate a user
+to access a MCP server and the MCP server determine whether to authorize the user to access its tools,
+resources, or any other capabilities.
+
+This document **does not discuss server-to-server authorization patterns**.
```

Comment:
This explicit scope is great, and I also think it's more reasonable to limit the OAuth2 related specifications described in the document to the **user authorization** scenario. However, the description in section 2.2.1 here is inconsistent with this scope agreement:

> Client Credentials: the client is another application (not a human)

Should the description in section 2.2.1 be removed? I believe Client Credentials should be considered in server-to-server authorization patterns.

Reply:
That's a fair call-out. I think the grant section alludes to the fact that _different grants_ may be supported between clients and servers. Ultimately, a client may choose to add support for client credentials too (as impractical as that may be for the general case), but this specification does not cover the implementation or architectural implications.